pandas/io/html.py

```diff
@@ -577,7 +577,7 @@ def _parse_tables(self, doc, match, attrs):
                 for elem in table.find_all(style=re.compile(r"display:\s*none")):
                     elem.decompose()
 
-            if table not in unique_tables and table.find(text=match) is not None:
```

Comment:
is this backwards compatible?

Reply:
It was introduced in version 4.4.0, and the minimum version in pandas is 4.8.2, so should be fine